agent/src/main/java/nu/studer/teamcity/buildscan/agent/BuildScanServiceMessageInjector.java

```diff
@@ -248,6 +254,13 @@ private String getMavenInvocationArgs(BuildRunnerContext runner) {
         return "-Dmaven.ext.class.path=" + asClasspath(extensionJars) + " " + asArgs(sysProps);
     }
 
+    private static boolean hasNoDevelocityExtensionApplied(BuildRunnerContext runner, MavenExtensions extensions) {
```

Comment:
The naming of `hasNoDevelocityExtensionApplied` confused me slightly. I only expected it to check for the `develocity-maven-extension`, not for the `gradle-enterprise-maven-extension` or a custom extension. That said, I can't think of a better name either that can encompass all three of these.